public/app/features/correlations/CorrelationsPage.test.tsx

```diff
@@ -0,0 +1,423 @@
+import { render, waitFor, screen, fireEvent } from '@testing-library/react';
+import { merge, uniqueId } from 'lodash';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { Observable } from 'rxjs';
+import { getGrafanaContextMock } from 'test/mocks/getGrafanaContextMock';
+
+import { DataSourcePluginMeta } from '@grafana/data';
+import { BackendSrv, FetchError, FetchResponse, setDataSourceSrv, BackendSrvRequest } from '@grafana/runtime';
+import { GrafanaContext } from 'app/core/context/GrafanaContext';
+import { contextSrv } from 'app/core/services/context_srv';
+import { configureStore } from 'app/store/configureStore';
+
+import { mockDataSource, MockDataSourceSrv } from '../alerting/unified/mocks';
+
+import CorrelationsPage from './CorrelationsPage';
+import { Correlation, CreateCorrelationParams } from './types';
+
+type DeepPartial<T> = T extends Array<infer U>
```

Comment:
Isn't it `import {DeepPartial} from "react-hook-form";`? If not - could it be defined in some core utilities in grafana-data?

Reply:
🙈 forgot to remove that one, thanks!

Reply:
163bb188c08f06e54f752cac8d174c2b8b12873e